src/components/CityButton.js

```diff
@@ -0,0 +1,67 @@
+import {useState} from 'react';
+import styled from 'styled-components';
+
+const options = ['All Cities', 'Berlin', 'Cologne', 'Frankfurt am Main', 'Hamburg', 'Munich'];
+
+export default function CityButton({selectedOption, selectEvents}) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const toggleOpen = () => setIsOpen(!isOpen);
+  const onEvent = city => {
+    selectEvents(city);
+    toggleOpen();
+  };
+
+  return (
+    <DropDownContainer>
+      <DropDownHeader onClick={toggleOpen}>{selectedOption || 'Choose City'}</DropDownHeader>
+      {isOpen && (
+        <DropDownListContainer>
+          <DropDownList>
+            {options.map(option => (
+              <ListItem onClick={() => onEvent(option)} key={Math.random()}>
```

Comment:
Instead of using `Math.random()` for/as your React key, you could make your options array an `array of objects`, where each object has an `id`. 

```
const locations = [
{
    id: 1,
    name: "Berlin"
 },
 {
    id: 1,
    name: "Köln"
 }
 ]
```

Reply:
Done!